core/pathresolver.go

```diff
@@ -29,29 +38,29 @@ func resolveRecurse(n *IpfsNode, p path.Path, depth int) (*merkledag.Node, error
 	// emerges when resolving just a <hash>. Is it meant
 	// to be an ipfs or an ipns resolution?
 
-	if strings.HasPrefix(p.String(), "/ipns/") {
+	if strings.HasPrefix(r.p.String(), "/ipns/") {
 		// if it's an ipns path, try to resolve it.
 		// if we can't, we can give that error back to the user.
-		seg := p.Segments()
+		seg := r.p.Segments()
 		if len(seg) < 2 || seg[1] == "" { // just "/ipns/"
-			return nil, fmt.Errorf("invalid path: %s", string(p))
+			return nil, fmt.Errorf("invalid path: %s", string(r.p))
 		}
 
 		ipnsPath := seg[1]
 		extensions := seg[2:]
-		respath, err := n.Namesys.Resolve(n.Context(), ipnsPath)
+		respath, err := r.n.Namesys.Resolve(r.ctx, ipnsPath)
```

Comment:
note: this is a bug that is prexisting, not caused by your code. just thought i'd mention it here since youre actively working in this area.

Try it out yourself, with no daemon running `ipfs ls /ipns/ipfs.git.sexy`